pkg/apis/apps.kubermatic/v1/application_definition.go

```diff
@@ -19,25 +19,32 @@ package v1
 import (
 	semverlib "github.com/Masterminds/semver/v3"
 
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type HelmCredentials struct {
-	// Username holds the ref and key in the secret for the username credential
-	Username GlobalSecretKeySelector `json:"username"`
+	// Username holds the ref and key in the secret for the username credential. Secret must exist in the namespace where
+	// KKP is installed.
+	Username corev1.SecretKeySelector `json:"username"`
 
-	// Password holds the ref and key in the secret for the Password credential
-	Password GlobalSecretKeySelector `json:"password"`
+	// Password holds the ref and key in the secret for the Password credential. Secret must exist in the namespace where
+	// KKP is installed.
+	Password corev1.SecretKeySelector `json:"password"`
 }
 
 type HelmSource struct {
-	// URl of the helm repository
+	// URl of the helm repository.
+	// It can be an HTTP(s) repository (e.g. https://localhost/myrepo) or on OCI repository (e.g. oci://localhost:5000/myrepo).
+	// +kubebuilder:validation:MinLength=1
```

Comment:
We should add a regex to validate this as well. Maybe something like 

```
+kubebuilder:validation:Pattern="(http|oci|https)://([a-zA-Z0-9~!@#$%^&*()_=+/?.:;',-]*)?"
```

Reply:
good point however I have used the more open regex `^(http|https|oci)://.+` because I am not totally sure that it's an URL. i doubt that but it could be an URI. Helm documentation only said  `OCI registry references are now always prefixed with oci://`